packages/cmf/src/sagas/http.js

```diff
@@ -213,33 +223,52 @@ export function* httpGet(url, config, options) {
 	return yield* wrapFetch(url, config, HTTP_METHODS.GET, undefined, options);
 }
 
-export const handleDefaultConfiguration = curry((defaultConfig, config) =>
-	mergeCSRFToken(defaultConfig)(config),
-);
+/**
+ * setDefaultHeader - define a default header to use with the saga http
+ *
+ * @param  {object} config key/value of header to apply
+ * @example
+ * import { setDefaultHeader } from '@talend/react-cmf/sagas/http';
+ * setDefaultHeader({headers: {
+ *  'Accept-Language': preferredLanguage,
+ * }});
+ */
+export function setDefaultConfig(config) {
+	defaultConfig = config;
+}
+
+/**
+ * getDefaultConfig - return the defaultConfig
+ *
+ * @return {object}  the defaultConfig used by cmf
+ */
+export function getDefaultConfig() {
+	return defaultConfig;
+}
 
 export default {
 	delete: httpDelete,
 	get: httpGet,
 	post: httpPost,
 	put: httpPut,
 	patch: httpPatch,
-	create(defaultConfig = {}) {
-		const configEnhancer = handleDefaultConfiguration(defaultConfig);
+	create(createConfig = {}) {
+		setDefaultConfig(createConfig);
```

Comment:
I don't think we should override the default config here.
You exposed and documented `setDefaultConfig()` which is explicit, so if we want to set default config, we should call it. Overriding it here will surely introduce unwanted behavior, where we want to pass custom config in `create()` but end up overriding config for all fetches.

Reply:
if I don't set setDefaultConfig, i have a breaking changes with data-streams. They use create with a default config.